batch.go

```diff
@@ -81,35 +90,67 @@ func (n *BatchNode) Count() int {
 
 func (n *BatchNode) Start() {
 	for _, b := range n.children {
-		b.(*QueryNode).Start()
+		switch b := b.(type) {
+		case *QueryNode:
+			b.Start()
+		case *FluxQueryNode:
+			b.Start()
+		default:
+			panic("BatchNode shouldn't be followed by anything except QueryNode or QueryFluxNode")
+		}
 	}
 }
 
 func (n *BatchNode) Abort() {
 	for _, b := range n.children {
-		b.(*QueryNode).Abort()
+		switch b := b.(type) {
+		case *QueryNode:
+			b.Abort()
+		case *FluxQueryNode:
+			b.Abort()
+		default:
+			panic("BatchNode shouldn't be followed by anything except QueryNode or QueryFluxNode")
+		}
 	}
 }
 
 type BatchQueries struct {
 	Queries            []*Query
+	FluxQueries        []*QueryFlux
 	Cluster            string
 	GroupByMeasurement bool
 }
 
 func (n *BatchNode) Queries(start, stop time.Time) ([]BatchQueries, error) {
 	queries := make([]BatchQueries, len(n.children))
-	for i, b := range n.children {
-		qn := b.(*QueryNode)
-		qs, err := qn.Queries(start, stop)
-		if err != nil {
-			return nil, err
-		}
-		queries[i] = BatchQueries{
-			Queries:            qs,
-			Cluster:            qn.Cluster(),
-			GroupByMeasurement: qn.GroupByMeasurement(),
+	for i, qn := range n.children {
+		switch qn := qn.(type) {
+		case *QueryNode:
+			qs, err := qn.Queries(start, stop)
+			if err != nil {
+				return nil, err
+			}
+			queries[i] = BatchQueries{
+				Queries:            qs,
+				Cluster:            qn.Cluster(),
+				GroupByMeasurement: qn.GroupByMeasurement(),
+			}
+
+		case *FluxQueryNode:
+			qs, err := qn.Queries(start, stop)
+			if err != nil {
+				return nil, err
+			}
+			queries[i] = BatchQueries{
+				FluxQueries: qs,
+				Cluster:     qn.Cluster(),
+				//GroupByMeasurement: qn.GroupByMeasurement(),
```

Comment:
Commented code - what does this do?

Reply:
Sorry that code was supposed to be deleted as GroupByMeasurement isn't meaningful for flux queries.